include/SFML/Graphics/Shape.hpp

```diff
@@ -127,6 +127,24 @@ class SFML_GRAPHICS_API Shape : public Drawable, public Transformable
     ////////////////////////////////////////////////////////////
     void setOutlineThickness(float thickness);
 
+    ////////////////////////////////////////////////////////////
+    /// \brief Set the limit on the ratio between miter length and outline thickness
+    ///
+    /// The edges of the shape's outline around sharp angles can meet
+    /// well beyond the outline thickness. When the ratio between the
+    /// miter length (distance between the outline's tip and the shape's
+    /// corner) and the outline thickness exceeds the given limit, then
+    /// the join is converted from a miter to a bevel.
+    /// The miter limit must be greater than or equal to 1.
+    /// By default, the miter limit is 4.
+    ///
+    /// \param miterLimit New miter limit
+    ///
+    /// \see getMiterLimit
+    ///
+    ////////////////////////////////////////////////////////////
+    void setMiterLimit(float miterLimit);
```

Comment:
Does this make more sense as a `sf::ConvexShape` feature? Mitering the outline will never be necessary for our prebuilt circle and rectangle shapes. I suppose you may want to miter the edges of a rectangle but that aesthetic consideration is not what we're setting out to solve.

Reply:
It would be more complex to implement and it would be missing from user class derived from `sf::Shape` so I think it is better to keep it in `sf::Shape`.